### DISCUSSION TITLE
How to use an existing Keycloak instance

### DISCUSSION THREAD

**Discussion by prmitic** [microcks/microcks#1388] [Q&A] [answered]
Hi,
I have an instance of Keycloak (22.0.11) on OpenShift cluster, which I use mostly for REST services. I would like to use that instance for microcks as well. 
I did not find any information about necessary steps for this situation.
What need to be changed in MicrocksInstall CRD for this to work.
Do I need to create realm on existing keycloak instance, or it will be created, as it is created with normal installations.

Also, microcks realm that can be found [here](https://github.com/microcks/microcks-keycloak/blob/main/microcks-realm/microcks-realm-sample.json), is for an older version of the keycloak (10.0.1) and import fails. Is there a newer version that can be imported to Keycloak.

**Comment by lbroudoux** [comment, marked as answer]
Hi @prmitic,

The documentation is here: https://microcks.io/documentation/references/configuration/security-config/#reusing-an-existing-keycloak with principles being introduced here: https://microcks.io/documentation/explanations/deployment-options/#2-partially-managed

You can check this sample of the CRD: https://github.com/microcks/microcks-ansible-operator/blob/master/deploy/samples/openshift-features-ext-sso.yaml that relies on an external Keycloak instance. Basically, you just set the `install: false` and provide a `url` and `privateUrl` for back-channel communication. 

The Keyclaok realms definition (full or addon) to use are located here: https://github.com/microcks/microcks/tree/master/install. They still referenced older version but are working correctly. 

Hope it helps,
